tools/windows-node-installer/pkg/cloudprovider/azure/vm.go

```diff
@@ -482,12 +494,108 @@ func (az *AzureProvider) constructStorageProfile(imageId string) (storageProfile
 }
 
 // getWorkerBackendPoolID gets the backend pool id of the worker node loadbalancer
-func (az *AzureProvider) getWorkerBackendPoolID() *string {
-	// We assume that BackendPoolID for worker vm follows the following format
-	// /subscriptions/$SUBSCRIPTID/resourceGroups/$INFRAID-rg/providers/Microsoft.Network/loadBalancers/$INFRAID/backendAddressPools/$INFRAID
-	var wbpi = fmt.Sprintf("/subscriptions/%s/resourceGroups/%[2]s-rg/providers/"+
-		"Microsoft.Network/loadBalancers/%[2]s/backendAddressPools/%[2]s", az.subscriptionID, az.infraID)
-	return &wbpi
+func (az *AzureProvider) getWorkerBackendPools(ctx context.Context) ([]network.BackendAddressPool, error) {
+	// Create a filter to get a resource of type virtualMachine which contains worker in its name
+	filter := "resourceType eq 'Microsoft.Compute/virtualMachines' and substringof('worker', name)"
+	// number of resources to retrieve
+	// https://learnk8s.io/kubernetes-node-size
+	// Kubernetes can technically accommodate 5000 nodes, considering that for optimal performance organizations limit at 500 worker nodes
+	top := int32(500)
+	vmResources, err := az.getResource(ctx, filter, top)
```

Comment:
Why do you need to introduce the resource client? Why can't you use the VM client? Which points to you should have added a comment explaining that here.

Reply:
In the vmClient, they need full vmName. We can use list() but list would return all the VMs in the resource group, we will then have to filter it to get the worker VMs using go code iterating through all the VM names.
getResource() does all that work for us. We just need to use the output in that case.

Reply:
Like I said you need to mention this in a comment.

Reply:
Will add a comment here